download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java

```diff
@@ -323,50 +356,95 @@ private void unpack(File outputFile) throws NoSuchArchiverException {
         outputFile.delete();
     }
 
-
-    private void doGet(File outputFile) throws Exception {
-        String urlStr = this.uri.toString();
-        String[] segments = urlStr.split("/");
-        String file = segments[segments.length - 1];
-        String repoUrl = urlStr.substring(0, urlStr.length() - file.length() - 1);
-        Repository repository = new Repository(repoUrl, repoUrl);
-
-        Wagon wagon = this.wagonManager.getWagon(repository.getProtocol());
-        if (readTimeOut > 0) {
-            wagon.setReadTimeout(readTimeOut);
-            getLog().info(
-                "Read Timeout is set to " + readTimeOut + " milliseconds (apprx "
-                    + Math.round(readTimeOut * 1.66667e-5) + " minutes)");
-        }
+	/**
+	 * Perform the get operation and outputs a bunch of debug messages
+	 * @param source the source to download data from
+	 * @param outputFile the output destination
+	 * @param parameters http parameters used for that request
+	 * @param headers http headers used for that request
+	 * @param retries number of times that query will be retried
+	 * @throws Exception
+	 */
+	private void doGet(final URL source, final File outputFile, final Map<String, String> parameters, final Map<String, String> headers, final int retries) throws Exception {
         ConsoleDownloadMonitor downloadMonitor = null;
-        if (this.session.getSettings().isInteractiveMode()) {
-            downloadMonitor = new ConsoleDownloadMonitor(this.getLog());
-            wagon.addTransferListener(downloadMonitor);
+        if (session.getSettings().isInteractiveMode()) {
+            downloadMonitor = new ConsoleDownloadMonitor(getLog());
         }
-
-        AuthenticationInfo authenticationInfo = new AuthenticationInfo();
-        if (StringUtils.isNotBlank(username)) {
```

Comment:
Yes, I think so. At the moment, don't change that; but if you know better way to handle the multiple authentication strategies, that'd be welcome as a separate PR.